portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -56,34 +63,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
  
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       Entity entity = new Entity("UserInfo", id);
-      entity.setProperty("id", id);
       entity.setProperty("nickname", nickname);
       // The put() function automatically inserts new data or updates existing data based on ID
       datastore.put(entity);
       // Direct the user to main page
       response.sendRedirect("/index.html");
     }
-
     
   }
 
-  private String createJson(String isLoggedIn, String url, String email, String nickname){
-      String json = "{ ";
-      json += "\"isLoggedIn\": \"" + isLoggedIn + "\", ";
-      json += "\"url\": \"" + url + "\", ";
-      json += "\"email\": \"" + email + "\", ";
-      json += "\"nickname\": \"" + nickname + "\"";
-      return json + " }";
-  }
-
   /**
    * Returns the nickname of the user with id, or empty String if the user has not set a nickname.
    */
   public static String getUserNickname(String id) {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     Query query =
         new Query("UserInfo")
-            .setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
+            .setFilter(new Query.FilterPredicate("__key__", Query.FilterOperator.EQUAL, KeyFactory.createKey("UserInfo", id)));
```

Comment:
You can replace `__key__` with Entity.KEY_RESERVED_PROPERTY

It's better to have less "Magic strings" if possible